packages/contracts/src/dollar/upgradeInitializers/DiamondInit.sol

```diff
@@ -63,6 +63,9 @@ contract DiamondInit is Modifiers {
         ls.stakingDiscountMultiplier = uint256(0.001 ether); // 0.001
         ls.blockCountInAWeek = 45361;
 
+        // reentrancy guard
+        _initReentrancyGuard();
+
```

Comment:
I thought the tests were already done, could you raise the reward?

Reply:
It's not just adding a test, if that were the case I would have done it
I couldn't find any test for **LibAppStorage.sol**, I would have to create the test file from 0

Reply:
It would be great if you could add 2 tests:
1. Check [here](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/test/diamond/DiamondInit.t.sol) that `store.reentrancyStatus` is set on `init()`
2. Take any function with the `nonReentrant()` modifier and check that it reverts [here](https://github.com/rotcivegaf/ubiquity-dollar/blob/issue-646/packages/contracts/src/dollar/libraries/LibAppStorage.sol#L62)

Reply:
Ok, but I'm afk, give me until monday

Reply:
I don't know if it was the most elegant way to solve it but it took me a long time